File: 2099-find-subsequence-of-length-k-with-the-largest-sum/2099-find-subsequence-of-length-k-with-the-largest-sum.kt
class Solution {
    fun maxSubsequence(nums: IntArray, k: Int): IntArray =
        nums.mapIndexed { index, num -> Pair(num, index) } 
            .sortedBy { it.first } 
            .takeLast(k) 
            .sortedBy { it.second }
            .map { it.first }
            .toIntArray()
}

File: 2099-find-subsequence-of-length-k-with-the-largest-sum/2099-find-subsequence-of-length-k-with-the-largest-sum.py
class Solution:
    def maxSubsequence(self, nums: List[int], k: int) -> List[int]:
        heap = []
        for n in nums:
            heapq.heappush(heap, n)
            if len(heap) > k:
                heapq.heappop(heap)
        cnt = Counter(heap)
        res = []
        for n in nums:
            if cnt[n] > 0:
                cnt[n] -= 1
                res.append(n)
        return res